app/src/main/java/com/paranoidandroid/journey/legplanner/fragments/DayPlannerFragment.java

```diff
@@ -146,7 +146,7 @@ public Day getSelectedDay() {
         List<Day> days = new ArrayList<>();
         Calendar calendar = new GregorianCalendar();
         calendar.setTime(leg.getStartDate());
-        while (calendar.getTime().before(leg.getEndDate()))
+        while (calendar.getTime().before(leg.getEndDate()) || calendar.getTime().equals(leg.getEndDate()))
```

Comment:
You could do:

    while (!calendar.getTime().after(leg.getEndDate()))

but this works fine.